ui/widgets/controlgroup.js

```diff
@@ -0,0 +1,242 @@
+/*!
+ * jQuery UI Controlgroup @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ */
+
+//>>label: Controlgroup
+//>>group: Widgets
+//>>description: Visually groups form control widgets
+//>>docs: http://api.jqueryui.com/controlgroup/
+//>>demos: http://jqueryui.com/controlgroup/
+//>>css.structure: ../themes/base/core.css, ../themes/base/controlgroup.css
+//>>css.theme: ../themes/base/theme.css
+
+( function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define( [
+			"jquery",
+			"../widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}( function( $ ) {
+
+return $.widget( "ui.controlgroup", {
+	version: "@VERSION",
+	defaultElement: "<div>",
+	options: {
+		disabled: null,
+		items: {
+			"button": "input[type=button], input[type=submit], input[type=reset], button, a",
+			"checkboxradio": "input[type='checkbox'], input[type='radio']",
+			"selectmenu": "select",
+			"spinner": ".ui-spinner-input",
+			"controlgroupLabel": ".ui-controlgroup-label"
+		},
+		direction: "horizontal",
+		excludeInvisible: true
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// To support the enhanced option in jQuery Mobile, we isolate DOM manipulation
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		this._callChildMethod( "destroy" );
+		this.childWidgets.removeData( "ui-controlgroup-data" );
+		this.element.removeAttr( "role" );
+	},
+
+	_initWidgets: function() {
+		var that = this,
+			childWidgets = [];
+
+		// First we iterate over each of the items options
+		$.each( this.options.items, function( widget, selector ) {
+			var widgets, labels,
+				options = {};
+
+			// We assume everything is in the middle to start because we can't determine
+			// first / last elements until all enhancments are done.
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+
+			// Make sure the widget actually exists and has a selector set
+			if ( $.fn[ widget ] && selector ) {
+
+				// Find instances of this widget inside controlgroup and set options
+				widgets = that.element.find( selector )[ widget ]( options );
+
+				widgets.each( function() {
+					var element = $( this ),
+
+					// Set data on the widget element pointing to the this.element of the widget
+					// and telling us what type of widget this is
+					widgetElement = element[ widget ]( "widget" ).data(
+						"ui-controlgroup-data",
+						element.data( "ui-" + widget.charAt(0).toUpperCase() + widget.slice(1) )
```

Comment:
@scottgonzalez @dmethvin reminds me of a great point. `$.data` is much much faster for this use case then `$.fn.data` because we don't want or care about html5 data attributes, and this is being done in a tight loop on create and refresh so we want to make sure this is efficient. Based on what @dmethvin is we want to use `$.data` we need to use camel case names.